src/public/client.ts

```diff
@@ -55,6 +55,42 @@ fetch('/api/mappings')
       new DisplayMap(e)
     })
   })
+const render = () => {
+  fetch('/api/mappings')
+    .then(r => r.json())
+    .then((data: Array<Mapping>) => {
+      data.reverse()
+      domainList.innerHTML = ''
+      data.forEach(e => {
+        if (e.domain && e.port) {
+          const eachDomain = document.createElement('div')
```

Comment:
look at what you did in line 54. `new DisplayMap(e)`
Reuse the classes you have already built.